tensorflow/core/kernels/mkl/mkl_matmul_ops_common.h

```diff
@@ -33,8 +33,17 @@ using mkldnn::stream;
 
 namespace tensorflow {
 
+#define L1_SIZE 32 * 1024
 typedef Eigen::ThreadPoolDevice CPUDevice;
 
+inline bool ExecuteSingleThreadedGemm(int m, int n, int k) {
+  // Ideally we would like to determine blocking and then come up with
+  // a heuristic but what we are targetting are very small models whose
+  // total size is < few L1's. So we will do this simple calculation
+  // to determine if the MM should be run on a single thread.
+  return ((sizeof(float) * (m * n + k * (m + n))) < L1_SIZE * 8);
```

Comment:
Just curious, why is `L1_SiZE` multiplied by 8? If it's some magic constant, please say so in the comment (in the code).